src/RenderItems.js

```diff
@@ -1,13 +1,51 @@
 import React, { Component } from 'react'
-import Data from './Data.js'
+// import Data from 'Data.js'
 import Item from './Item.js'
 
+
+
 export default class RenderItems extends Component {
+    state = {
+        filter: ''
+    }
+
+    keySet = [...new Set(this.props.Data.map(item => item.keyword))]
+    hornSet = [...new Set(this.props.Data.map(item => item.horns))]
+
+
+    handleChange = e => {
+        this.setState({
+            filter: e.target.value,
+
+        });
+    }
     render() {
+
+        const filteredAnimals = this.props.Data.filter((animal) => {
+            if (!this.state.filter) return true;
+
+
+            if (animal.keyword === this.state.filter || animal.horns === +this.state.filter) return true;
```

Comment:
There's something called method chaining, this is something you could refactor and experiment with. You could have a .filter followed by another .filter.